be/src/agent/heartbeat_server.cpp

```diff
@@ -207,7 +207,9 @@ StatusOr<HeartbeatServer::CmpResult> HeartbeatServer::compare_master_info(const
     if (curr_master_info->cluster_id == -1) {
         LOG(INFO) << "Received first heartbeat. updating cluster id";
         // write and update cluster id
-        RETURN_IF_ERROR(_olap_engine->set_cluster_id(master_info.cluster_id));
+        if (_olap_engine->get_need_write_cluster_id()) {
```

Comment:
merge this if condition together with line 207. LOG line in line 208 should be moved into this need_write_cluster_id() if clause.